pkg/cmd/server/origin/run_components.go

```diff
@@ -265,9 +288,15 @@ func (c *MasterConfig) RunDeploymentController() {
 	if err != nil {
 		glog.Fatalf("Unable to initialize deployment controller: %v", err)
 	}
+
+	port, err := kubeserver.GetKubernetesMasterPort(&c.Options)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	master := fmt.Sprintf("https://kubernetes.default.svc.%s:%d", getDNSDomain(&c.Options, false), port)
 	// TODO eliminate these environment variables once service accounts provide a kubeconfig that includes all of this info
 	env := clientcmd.EnvVars(
-		kclientConfig.Host,
+		master,
```

Comment:
That sets the EnvVars for the deployer pod. The deployer pod is running as a pod and should access the master the same way all other pods access the master (imho) via the kubernetes service address. The previous setting "kclientConfig.Host" is the address the master components use to talk to each other... it should be able to be "localhost". That is another reason why the deployer pod shouldn't use the same address.